### ISSUE TITLE
No matching audio stream!

### ISSUE THREAD

**Issue by Tuizi** [JimmyBoh/pully#22]
I get the error: "No matching audio stream!" when trying to save with the preset "mp3" this video https://www.youtube.com/watch?v=RKpjqddcuy0.

Any idea why>

**Comment by Tuizi** [comment]
Yes it works, it just few video where I get this issue. Maybe 10% of all video I try to process, I can give you other example if you want

**Comment by JimmyBoh** [comment]
I think I need to add some more debugging functionality to really see what the issue might be. I would think that there is something going on with those videos, not necessarily this library. But I could be wrong! Under the covers I use `ytdl-core`, which does the actual parsing of various media types and downloading. I will try to take a look at this sometime this week.

 (In the meantime if mp3 versions are critical try downloading the whole video and then use another program to extract just the audio (ffmpeg would be perfect)

**Comment by Tuizi** [comment]
Ok keep me updated and I will try the full video + ffmpeg process

**Comment by tkissing** [comment]
Same problem here. I tried about a dozen videos from https://www.youtube.com/channel/UCEGJQkfFh4wk2HYtPLInnPA/videos and had that error with each of them :(

**Comment by JimmyBoh** [comment]
Hmm, there should be a common ground when it comes to the raw data coming back. We can try adding more logging when it fails and see if we can determine anything.

**Comment by dolanmiu** [comment]
Failed for this video too:

https://www.youtube.com/watch?v=VfnYmGSE1aY

Why does this happen?

**Comment by nakleiderer** [comment]
I was having a similar issue, but with video streams. All video formats were being filtered out because the format.type was undefined. This issue was related to the raw field changing from `type` to `mimeType`. This was addressed in https://github.com/jimbuck/pully-core/commit/1b2d4275c564235b18d356815f82d6facc068b64 when https://github.com/jimbuck/pully-core/blame/master/src/index.ts#L42 was changed. Updating to the latest pully-core resolved the issue for me. I had to do it the dirty way by reaching into node_modules/pully and installing the latest version manually. It looks like pully-core will be upgraded in the next release due to this commit https://github.com/jimbuck/pully/commit/c491a50ebc019edb5395bc8fab10f3fa999fed60

**Comment by jimbuck** [comment]
@nakleiderer You are absolutely correct. A few things changed in the underlying APIs so I am working through those now. Hopefully I'll get pully updated and released soon.

**Comment by GreysonR** [comment]
I found out that in `analyzer.ts` (or \node_modules\pully\dist\lib\analyzer.js if you installed from npm), in the `_createVideoFilter` and `_createAudioFilter` functions, you need to change every `format.type` to `format.raw.mimeType`.
